tiledb/sm/cpp_api/dimension.h

```diff
@@ -258,6 +258,12 @@ class Dimension {
   template <typename T>
   T tile_extent() const {
     impl::type_check<T>(type(), 1);
+    auto te = _tile_extent();
+    if (te == NULL)
+    {
+      T value = T();
+      return value;
```

Comment:
do `return T()` to avoid the extra assignment-operator on L264.

Reply:
thanks for suggestion